src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -0,0 +1,794 @@
+        try {
+            if (canCreateChildrenHandler == null) {
+                return true;
+            }
+
+            return canCreateChildrenHandler.test(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canCreateChildrenHandler");
+        }
+    }
+
+    /**
+     * Determines if this file can be created as new directory.
+     *
+     * @return <tt>true</tt> if this file can (probably) be created as directory
+     */
+    public boolean canCreateAsDirectory() {
+        try {
+            if (exists() && !isDirectory()) {
+                return false;
+            }
+
+            if (canCreateDirectoryHandler != null) {
+                return canCreateDirectoryHandler.test(this);
+            }
+
+            return createDirectoryHandler != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canCreateDirectoryHandler");
+        }
+    }
+
+    /**
+     * Tries to create this file as directory.
+     *
+     * @return <tt>true</tt> if the operation was successful, <tt>false</tt> otherwise
+     */
+    public boolean tryCreateAsDirectory() {
+        try {
+            if (exists()) {
+                return isDirectory();
+            }
+
+            if (createDirectoryHandler == null) {
+                return false;
+            }
+
+            return createDirectoryHandler.apply(this);
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "createDirectoryHandler");
+        }
+    }
+
+    /**
+     * Creates this file as new directory.
+     *
+     * @throws HandledException if the file cannot be created as new directory
+     */
+    public void createAsDirectory() {
+        if (!tryCreateAsDirectory()) {
+            throw Exceptions.createHandled()
+                            .withNLSKey("VirtualFile.cannotCreateAsDirectory")
+                            .set("file", path())
+                            .handle();
+        }
+    }
+
+    /**
+     * Determines if an {@link OutputStream} can (probably) be created for this file.
+     * <p>
+     * Note that {@link #isWriteable()} is probably the better check.
+     *
+     * @return <tt>true</tt> if an output stream can be created, <tt>false</tt> otherwise
+     */
+    public boolean canCreateOutputStream() {
+        try {
+            if (canProvideOutputStream != null) {
+                return canProvideOutputStream.test(this);
+            }
+
+            return outputStreamSupplier != null;
+        } catch (Exception e) {
+            throw handleErrorInCallback(e, "canProvideOutputStream");
+        }
+    }
+
+    /**
+     * Tries to create an output stream to write to the file.
+     *
+     * @return an output stream to provide the contents of the child or an empty optional if no output stream can be created
+     */
+    public Optional<OutputStream> tryCreateOutputStream() {
+        try {
+            if (outputStreamSupplier == null) {
```

Comment:
I think inverting this if check would improve readability/flow and would be consistent to the other methods in this class